healthcheck/dial.go

```diff
@@ -64,9 +64,10 @@ func sockaddrToString(sa syscall.Sockaddr) string {
 
 // dialTCP dials a TCP connection to the specified host and sets marking on the
 // socket. The host must be given as an IP address. A mark of zero results in a
-// normal (non-marked) connection.
-func dialTCP(network, addr string, timeout time.Duration, mark int) (nc net.Conn, err error) {
-	host, port, err := net.SplitHostPort(addr)
+// normal (non-marked) connection. For tunneling mode healthcheck, source IP
```

Comment:
I still don't get the reason why source IP is only needed for Tunnel mode while DSR mode doesn't. For DSR, the reason I guess is that kernel picks the source address based on backend IP not the service IP, so backend can send back the response. In Tunnel mode, how kernel will pick a source IP? based on backend IP or service IP? If it's backend IP, it should still work.

Reply:
@liuyuan10 That is a good question. I tend to agree ipvs DR (or DSR in seesaw) should be the same as TUN mode, in terms of picking source address. I don't have a test environment right now for checking it further, but in my past experience, if not explicitly set, TUN mode will picking a wrong IP in my setup.

I'll redo the test and get it back to you. Meanwhile, since this PR contains too many commits, do you think we can hold off changes regarding source IP selection, basically following two commits:

- Allow tunnel mode healthcheck to use specified source IP 
- Use engine node address for tunnel mode healthcheck source address

And try if we can first get the rest reviewed first? If this approach works with you, I'll file another PR for the rest commits and leave this PR open for source IP selection changes. Let me know. Thanks!

Reply:
> if not explicitly set, TUN mode will picking a wrong IP in my setup.

Guess you have multiple IPs in your setup. Could you play with the routing table a bit to find a clue how kernel decide the inner source address in TUN mode?

> And try if we can first get the rest reviewed first? If this approach works with you, I'll file another PR for the rest commits and leave this PR open for source IP selection changes. Let me know. Thanks!

I'm fine with that. Could you squash the rest of commits and drop those two from the PR? I'm not a fan of many commits for one topic. But you can take your own judgement.

Reply:
> Guess you have multiple IPs in your setup. Could you play with the routing table a bit to find a clue how kernel decide the inner source address in TUN mode?

Sure, will do.

> I'm fine with that. Could you squash the rest of commits and drop those two from the PR? I'm not a fan of many commits for one topic. But you can take your own judgement.

Those two commits dropped. Also squashed a few. Will push for the next update now.